### PR TITLE
EXL-390 - Made category field required

### PR DESCRIPTION
**Pull request by zsso**
- Added functionality to check if installation date field is higher the effective date field
- Added alert in date from CSV is lower than minimum required effective date
- Added "Yesterday and Today" buttons to installation date field
- Added functionality to make installation date field equal with effective date, unless it was manually changed
- Disabled every filed on ticket form when ticket is closed
- Bug fixes

### CODE REVIEW

File: plugin/inc/ticket.class.php [hunk 1]
```diff
@@ -1834,12 +1852,12 @@ public static function moveCartridges(Ticket $item): void
                 $ticket->customfields->fields['total2_color_field'] ?? null,
                 $ticket->customfields->fields['cartridge_install_date_field'] ?? $ticket->customfields->fields['effective_date_field']
             );
-            if (abs($install_result) != $cartridge_item['cartridges_id']) {
+            if (abs(intval($install_result)) != $cartridge_item['cartridges_id']) {
```
Comment (hupupu) [inline plugin/inc/ticket.class.php:1855]: Can `$install_result` be something else than an int?

File: plugin/templates/pages/support/ticket.html.twig [hunk 2]
```diff
@@ -17,8 +17,11 @@
         'locked_fields': item.getLockedFields(),
         'full_width': true,
         'fields_template': iServiceTicketTemplate,
+        'disabled': isClosed,
     } %}
 
+    {% set canupdate = false %}
+
```
Comment (hupupu) [inline plugin/templates/pages/support/ticket.html.twig:24]: Is this really needed here or is just a leftover code?

**Review by hupupu — APPROVED** [review]

File: plugin/inc/ticket.class.php (hunk 1 above)
Comment (zsso) [reply plugin/inc/ticket.class.php:1855]: Yes, if there is an error, string is returned, error message.

File: plugin/templates/pages/support/ticket.html.twig (hunk 2 above)
Comment (zsso) [reply plugin/templates/pages/support/ticket.html.twig:24]: Leftover :(

**Pull request merged**